packages/hoppscotch-app/components/http/Body.vue

```diff
@@ -80,19 +91,45 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from "@nuxtjs/composition-api"
+<script setup lang="ts">
+import { Ref, watch, ref } from "@nuxtjs/composition-api"
+import { HoppRESTHeader } from "@hoppscotch/data"
+import { RequestOptionTabs } from "./RequestOptions.vue"
 import { useStream } from "~/helpers/utils/composables"
-import { restContentType$, setRESTContentType } from "~/newstore/RESTSession"
 import { knownContentTypes } from "~/helpers/utils/contenttypes"
+import {
+  restContentType$,
+  restHeaders$,
+  setRESTContentType,
+  setRESTHeaders,
+} from "~/newstore/RESTSession"
 
-export default defineComponent({
-  setup() {
-    return {
-      validContentTypes: Object.keys(knownContentTypes),
+const emit = defineEmits(["changeTab"])
 
-      contentType: useStream(restContentType$, null, setRESTContentType),
-    }
+const validContentTypes = Object.keys(knownContentTypes)
+const contentType = useStream(restContentType$, null, setRESTContentType)
+
+// The functional headers list (the headers actually in the system)
+const headers = useStream(restHeaders$, [], setRESTHeaders) as Ref<
+  HoppRESTHeader[]
+>
+
+const overridenContentType = ref<string | null>(null)
+
+watch(
```

Comment:
Can't `overridenContentType` just be a computed property ?
Also, during the rewrite, try using `fp-ts` to build the result.